docs/appnotes/0009-storage-label-format.md

```diff
@@ -0,0 +1,97 @@
+So the signalling of these properties must be optional.
+Availability zones may also be semi-randomised in their naming to aid in evenly distributing load across cloud infrastructure.
+Zone naming is consistent within cloud accounts, though.
+Co-locating within a zone can be beneficial to performance.
+But this is only possible if a client knows it is within the same account as the server.
+It must, then, be possible for a client to either use or ignore availability zone based on whether it is within the same cloud account.
+This gives us provider, region, and availability zone as the important information that can be used to signal the location of storage.
+
+In addition to location, storage type is important for inferring the properties of the storage.
+Unfortunately, it is not possible or practical to directly compare similar types of products.
+For example, cloud providers may provide Object Storage.
+But they may provide multiple types of Object Storage.
+These may have vastly different properties.
+And there is often no universal naming conventions to describe these products in a way that is comparable between cloud providers.
+So while we need to signal this, we are not able to provide universal naming conventions.
+
+Finally, a more generic "store name" parameter is useful for human identification of stores, and for distinguishing stores which are otherwise identical.
+
+In summary; we must signal storage provider, region, availability zone, storage type, and store name.
+We cannot, unfortunately, signal any of these in consistently named and universally comparable ways.
+But we can specify a schema which allows a client to consistently decide which pieces of information are important to it.
+
+## Content
+
+Given all of the above, this Application Note recommends the following naming convention for the `get_urls` `label` parameter on flow segments:
+
+```text
+<provider>.<region[optional]>.<availabilityZone[optional]>:<storeType>:<storeName>
+```
+
+This can be represented more formally with the following Python-compatible regex:
+
+```regex
+^(?P<provider>[A-Za-z0-9\-\_]+)(.(?P<region>[A-Za-z0-9\-\_]+)(.(?P<availabilityZone>[A-Za-z0-9\-\_]+))?)?:(?P<storeType>[A-Za-z0-9\-\_]+):(?P<storeName>[A-Za-z0-9\-\_]+)$
+```
+
+An example use of this would be:
+
+```text
+example-cloud-provider.eu-west-1.a:example-storage-product:example-store-name
+```
+
+An example use of this without an availability zone would be:
+
+```text
+example-cloud-provider.eu-west-1:example-storage-product:example-store-name
+```
+
+An example use of this without a region would be:
+
+```text
+example-cloud-provider:example-storage-product:example-store-name
+```
+
+The parameters `provider`, `region`, `availabilityZone`, and `storeType` should use the machine readable values as provided by the cloud/storage vendor.
+The intention of this approach is to allow consistent values to be used without enumerating common/possible values in TAMS.
```

Comment:
Sadly I think a level of enumeration is going to be necessary somewhere for consistency.  For example is AWS listed as 'amazon' or 'aws'?  Regions is probably the simplest for public clouds as you can take their naming convention, eg 'eu-west-1', however the same will be necessary at the storage product level for example to describe S3 Express One Zone in a common way.

These probably will ultimately need to end up in an app note somewhere similar to the tags list, so not part of the API spec or validation

Reply:
Hm. Is there no consistency in the likes of ARN's for aws? I'd expected inconsistency in documentation, but hoped for consistency in configuration.

Reply:
Apologies, I missed the line directly above stating where the values should come from.  While the AWS ARN structure is well defined there are some oddities that stand out like s3 buckets not having the region defined.  Similarly while the majority of services define the AZ in the format eu-west-1a, I have seen some exceptions to this.  Ultimately it might be worth either pointing to where the enumerations come from, or alternatively giving some examples for a few of the most common storage platforms.

Reply:
I've expanded explanations and added examples in c46f8c3